cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"path"
+
+	certphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/certs"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	certutil "k8s.io/kubernetes/pkg/util/cert"
+)
+
+const KubernetesDirPermissions = 0700
+
+// This function is called from the main init and does the work for the default phase behaviour
+func CreateAdminAndKubeletKubeConfig(masterEndpoint, pkiDir, outDir string) error {
+	// Parse the certificate from a file
+	caCertPath := path.Join(pkiDir, "ca.pem")
+	caCerts, err := certutil.CertsFromFile(caCertPath)
+	if err != nil || len(caCerts) == 0 {
+		return fmt.Errorf("couldn't read or parse the ca cert file %s: %v", caCertPath, err)
+	}
+	// We are only putting one certificate in the CA certificate pem file, so it's safe to just use the first one
+	caCert := caCerts[0]
+
+	// Parse the rsa private key from a file
+	caKeyPath := path.Join(pkiDir, "ca-key.pem")
+	priv, err := certutil.PrivateKeyFromFile(caKeyPath)
+	if err != nil {
+		return fmt.Errorf("couldn't read or parse the ca private key file %s: %v", caCertPath, err)
+	}
+	var caKey *rsa.PrivateKey
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		caKey = k
+	case *ecdsa.PrivateKey:
+		return fmt.Errorf("the CA private key file %s isn't in RSA format", caCertPath)
```

Comment:
added a TODO for this and a comment why it can't be supported right away
deferring to the future